### ISSUE TITLE
`signature-in-docstring` (`D402`) - false positive on usage example

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#16824]
### Summary

```py
def do_while(condition: Callable[[], bool]) -> Generator[bool]:
    """usage: `for _ in do_while(...)`.""" # error: First line should not be the function's signature
```

[playground](https://play.ruff.rs/8c8144fb-c8c3-4501-8d26-9afe3dffd08d)

you could argue that the docstring shouldn't have an example as the first line, but that's a separate issue. i think `signature-in-docstring` should not be reported in this case because it's not a signature. perhaps it should only be reported if the function name is at the start of the docstring [like the example in the docs](https://docs.astral.sh/ruff/rules/signature-in-docstring/#example)

**Comment by ntBre** [comment]
Hmm, this seems to be consistent with the [implementation](https://github.com/PyCQA/pydocstyle/blob/8d0cdfc93e86e096bb5753f07dc5c7c373e63837/src/pydocstyle/checker.py#L562) of the upstream rule, and we also have a similar test case:

https://github.com/astral-sh/ruff/blob/682ba383cb9e5f1369ce3ccb6ff71f7b65ce72a4/crates/ruff_linter/resources/test/fixtures/pydocstyle/D402.py#L7-L8

But I also kind of agree that it's a pretty imprecise check.
